src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -294,6 +294,15 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
             return $a == $b ? 0 : 1;
         });
 
+        array_walk_recursive($extra, $caster = static function (&$v) use (&$caster) {
+            if ($v instanceof \stdClass) {
+                $v = (array) $v;
+                array_walk_recursive($v, $caster);
+            } elseif (null !== $v) {
```

Comment:
To account for Tobion's feedback, we should change this to `} elseif (\is_object($v)) {`

Reply:
Done